java/apirag-openai/src/main/java/com/datasqrl/ai/models/openai/OpenAiChatProvider.java

```diff
@@ -72,14 +70,14 @@ public GenericChatMessage chat(String message, Map<String, Object> context) {
       GenericChatMessage genericResponse = session.addMessage(responseMessage);
       ChatFunctionCall functionCall = responseMessage.getFunctionCall();
       if (functionCall != null) {
-        FunctionValidation<ChatMessage> fctValid = this.validateFunctionCall(functionCall);
+        FunctionValidation<ChatMessage> fctValid = session.validateFunctionCall(functionCall);
         if (fctValid.isValid()) {
```

Comment:
I think we want another method in ChatSession that encapsulates this entire if block in a method call since that is identical for all implementations.

Reply:
Ah yes, I will add a method and abstract this